calicoctl/commands/resources.go

```diff
@@ -182,12 +183,34 @@ func executeConfigCommand(args map[string]interface{}, action action) commandRes
 
 	// Now execute the command on each resource in order, exiting as soon as we hit an
 	// error.
+	export := argutils.ArgBoolOrFalse(args, "--export")
+	nameSpecified := argutils.ArgStringOrBlank(args, "<NAME>")
 	for _, r := range resources {
 		res, err := executeResourceAction(args, client, r, action)
 		if err != nil {
 			results.err = err
 			break
 		}
+
+		for i, _ := range res {
+			// Remove the cluster specific metadata if the "--export" flag is specified
+			if export {
+				if nameSpecified == "" {
```

Comment:
line 195 to 202 can be simplified to:
```go
if exported && nameSpecified != "" {
     for i, _ := range res {
          rom := res[i].(v1.ObjectMetaAccessor).GetObjectMeta()
          ...
          ...
     }
}
```

Reply:
oops wrote it before I had tested if kubectl gives a log or anything for specifying `--export` without a name. I'll fix that up.